client/components/RoomListItem.jsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+
+const RoomListItem = React.createClass({
+  mixins: [ReactMeteorData],
+
+  getMeteorData () {
+    const messagesHandle = Meteor.subscribe("chatterMessages", {
+      roomId: this.props.room._id,
+      messageLimit: 30
+    });
+
+    const countHandle = Meteor.subscribe("chatterUserRooms", {
+      roomId: this.props.room._id
+    });
+
+    const subsReady = messagesHandle.ready() && countHandle.ready();
+
+    let message = "no messages yet",
+        avatar = "http://localhost:3000/packages/jorgeer_chatter-semantic/public/images/default.jpg",
+        timeAgo = "",
+        count = 0;
+
+    if (subsReady) {
+      const checkCount = Chatter.UserRoom.findOne({roomId: this.props.room._id, userId: this.props.chatterUser._id});
+
+      count =  typeof checkCount === 'undefined' ? -1 : checkCount.count;
+
+      const lastMessage = Chatter.Message.findOne({roomId: this.props.room._id }, {sort: {createdAt: -1, limit: 1}});
+      if (typeof lastMessage != 'undefined') {
+        message = lastMessage.message;
+        timeAgo = lastMessage.timeAgo();
+        avatar = lastMessage.userAvatar;
+      }
+    }
+
+    return {
+      message,
+      avatar,
+      timeAgo,
+      count
+    }
+  },
+
+  render() {
```

Comment:
de-structure this.props so you don't have to repeat it so many times and get shorter code.
```
const {
  goToRoom,
  room
  etc.
} = this.props;

```